crypto/cms/cms_sd.c

```diff
@@ -875,6 +889,13 @@ int CMS_SignerInfo_sign(CMS_SignerInfo *si)
         si->pctx = pctx;
     }
 
+    if (md_name == NULL) {
```

Comment:
md_name can never be NULL with the current code with your #if 0 commenting out the only way it could be set

Reply:
Sure. I'm trying to figure out why the tests fail for 32bit windows...